pkg/controllers/externalsecret/externalsecret_controller_template.go

```diff
@@ -113,14 +127,15 @@ func setMetadata(secret *v1.Secret, es *esv1beta1.ExternalSecret) error {
 		delete(secret.ObjectMeta.Annotations, key)
 	}
 
+	// if no template is defined, copy labels and annotations from the ExternalSecret
+	// otherwise copy from the template
 	if es.Spec.Target.Template == nil {
 		utils.MergeStringMap(secret.ObjectMeta.Labels, es.ObjectMeta.Labels)
 		utils.MergeStringMap(secret.ObjectMeta.Annotations, es.ObjectMeta.Annotations)
-		return nil
+	} else {
```

Comment:
I would keep the return nil here and put the rest below this instead of an else. That's a none zero value operation to run an else section.

Reply:
I have changed this in https://github.com/external-secrets/external-secrets/pull/4086/commits/62b658dc99f00969866ec050b08cd47d9e798db9